src/main/scala/org/broadinstitute/hail/utils/richUtils/RichMultiArray2Numeric.scala

```diff
@@ -0,0 +1,30 @@
+package org.broadinstitute.hail.utils.richUtils
+
+import org.broadinstitute.hail.utils.MultiArray2
+
+class RichMultiArray2Numeric[T](val ma: MultiArray2[T]) extends AnyVal {
+
+  def addElementWise(other: MultiArray2[T])(implicit ev: T => scala.math.Numeric[T]#Ops): MultiArray2[T] = {
```

Comment:
I wonder how this `ev` stuff looks in bytecode.  If this is performance critical, specializing the type might be a big improvement vs. a (couple?) virtual function calls.